src/utils/getUserRewards.ts

```diff
@@ -78,63 +86,73 @@ export const getUserRewards = async (
     claimData,
   };
 };
-export const userBalancesToUnclaimedTokens = (
+export const userBalancesToUnclaimedTokens = async (
   balances: UserBalance[],
-  currentTimestamp: BigNumberish
+  currentTimestamp: BigNumberish,
+  provider: providers.Provider
 ) => {
-  return balances
-    .map((b) => {
+  return Promise.all(
+    balances.map(async (b) => {
       let accumulated = b.accumulatedMorpho;
-      const supplyIndex = computeSupplyIndex(b.market, currentTimestamp);
+      const supplyIndex = await computeSupplyIndex(b.market, currentTimestamp, provider);
       accumulated = accumulated.add(
         getUserAccumulatedRewards(supplyIndex, b.userSupplyIndex, b.underlyingSupplyBalance)
       );
-      const borrowIndex = computeBorrowIndex(b.market, currentTimestamp);
+      const borrowIndex = await computeBorrowIndex(b.market, currentTimestamp, provider);
       accumulated = accumulated.add(
         getUserAccumulatedRewards(borrowIndex, b.userBorrowIndex, b.underlyingBorrowBalance)
       );
       return accumulated;
     })
-    .reduce((a, b) => a.add(b), BigNumber.from(0));
+  ).then((r) => r.reduce((a, b) => a.add(b), BigNumber.from(0)));
 };
 
 const getUserAccumulatedRewards = (marketIndex: BigNumber, userIndex: BigNumber, userBalance: BigNumber) => {
   if (userIndex.gt(marketIndex)) return BigNumber.from(0);
```

Comment:
I think i'll need a crash course about indexes and how we handle it in the dapp but what i understand is that `userIndex` is the index when the user "entered" the market whereas `marketIndex` is the current index of the maket . Am I right? Then, since indexes are supposed to only increase, how can this happen?